joomla/joomla-cms
# [4.1] Misc improvements for Task\TaskOption

### Summary of Changes
- Add `@property-reads` docs for better IDE/static analysis support.
- Refactor `TaskOption::type` to `TaskOption::id`

### Testing Instructions
- Create a new Scheduled Task with a short interval-based schedule (1 minute).
- Let it run organically at planned time (should be a minute after created).
- Run it manually from the button in the Scheduled Tasks manager.


### Actual result BEFORE applying this Pull Request
- Everything works as expected; but:  


- TaskOption::type doesn't make sense.
- Poor IDE support for $taskOption->{property} since they're accessed only through TaskOption::__get().

### Expected result AFTER applying this Pull Request
- Everything works as expected; but:  

- TaskOption::id makes more sense.
- IDE doesn't complain about valid $taskOption->{property{ reads.


### Documentation Changes Required
\-
